python_modules/dagster/dagster/core/execution_context.py

```diff
@@ -245,3 +249,171 @@ def __new__(cls, run_id=None, tags=None, event_callback=None, loggers=None):
+
+    @abstractmethod
+    def has_event_callback(self):
+        pass
+
+    @abstractproperty
+    def environment_config(self):
+        pass
+
+    @abstractproperty
+    def context(self):
+        pass
+
+    @abstractproperty
+    def config(self):
+        pass
+
+    @abstractproperty
+    def step(self):
+        pass
+
+    @abstractproperty
+    def solid_def(self):
+        pass
+
+    @abstractproperty
+    def solid(self):
+        pass
+
+    @abstractproperty
+    def pipeline_def(self):
+        pass
+
+    @abstractproperty
+    def resources(self):
+        pass
+
+    @abstractproperty
+    def log(self):
+        pass
+
+
+class WithLegacyContext(six.with_metaclass(ABCMeta)):  # pylint: disable=no-init
+    @abstractproperty
+    def legacy_context(self):
+        pass
+
+    def has_tag(self, key):
+        check.str_param(key, 'key')
+        return self.legacy_context.has_tag(key)
+
+    def get_tag(self, key):
+        check.str_param(key, 'key')
+        return self.legacy_context.get_tag(key)
+
+    @property
+    def run_id(self):
+        return self.legacy_context.run_id
+
+    @property
+    def event_callback(self):
+        return self.legacy_context.event_callback
+
+    @property
+    def has_event_callback(self):
+        return self.legacy_context.has_event_callback
+
+    @property
+    def environment_config(self):
+        return self.legacy_context.environment_config
+
+    @property
+    def log(self):
+        return DagsterLog(self.legacy_context)
+
+    @property
+    def resources(self):
+        return self.legacy_context.resources
+
+    @property
+    def persistence_policy(self):
+        return self.legacy_context.persistence_policy
+
+    @property
+    def events(self):
+        return self.legacy_context.events
+
+    @property
+    def loggers(self):
+        return self.legacy_context.loggers
+
+
+class TransformExecutionContext(
+    WithLegacyContext,
+    ITransformExecutionContext,
+    namedtuple('_TransformExecutionContext', 'context__ config__ step__ pipeline_def__'),
+):
+    '''An instance of TransformExecutionContext is passed every solid transform function.
+
+    Attributes:
+
+        context (ExecutionContext): Context instance for this pipeline invocation
+        config (Any): Config object for current solid
+    '''
+
+    def __new__(cls, legacy_context, config, step, pipeline_def):
+        from dagster.core.definitions.pipeline import PipelineDefinition
+        from dagster.core.execution_plan.objects import ExecutionStep
+
+        return super(TransformExecutionContext, cls).__new__(
+            cls,
+            check.inst_param(legacy_context, 'legacy_context', LegacyRuntimeExecutionContext),
+            config,
+            check.inst_param(step, 'step', ExecutionStep),
+            check.inst_param(pipeline_def, 'pipeline_def', PipelineDefinition),
+        )
+
+    @property
+    def context(self):
+        # Deprecated
+        return self.context__
```

Comment:
This would be a great place to insert a call to `warnings.warn`.

Reply:
good idea. done.